File: cliente/src/app/modulos/capturas/modelos-capturas.ts
export type EstadoCaptura = 'Pendiente' | 'EnProceso' | 'Completada' | 'Fallida' | 'Cancelada';

export type OrigenVersion = 'CapturaSsh' | 'CapturaNetconf' | 'Archivo';

export interface CapturaResumen {
  id: number;
  dispositivoId: number;
  dispositivo: string;
  usuarioSolicitante: string | null;
  disparador: string;
  medio: string;
  estado: EstadoCaptura;
  fecha: string;
  error: string | null;
  versionId: number | null;
}

export interface VersionConfiguracionResumen {
  id: number;
  dispositivoId: number;
  dispositivo: string;
  capturaId: number;
  numero: number;
  origen: OrigenVersion;
  hash: string;
  capturadaEn: string;
  comentario: string | null;
  estado: 'Activa' | 'Retirada';
  estable: boolean;
  usuarioSolicitante: string | null;
}

export interface VersionConfiguracionDetalle extends VersionConfiguracionResumen {
  contenido: string;
}

export interface CargaArchivoResultado {
  captura: CapturaResumen;
  version: VersionConfiguracionResumen;
}

File: servidor/fuente/RedDiff.Aplicacion/Capturas/ModelosCapturas.cs
namespace RedDiff.Aplicacion.Capturas;

public sealed record CargarArchivoConfiguracionSolicitud(
    long DispositivoId,
    string NombreArchivo,
    byte[] Contenido,
    string? Comentario);

public sealed record CapturarConfiguracionRemotaSolicitud(long DispositivoId);

public sealed record CapturaResumen(
    long Id,
    long DispositivoId,
    string Dispositivo,
    string? UsuarioSolicitante,
    string Disparador,
    string Medio,
    string Estado,
    DateTimeOffset Fecha,
    string? Error,
    long? VersionId);

public sealed record VersionConfiguracionResumen(
    long Id,
    long DispositivoId,
    string Dispositivo,
    long CapturaId,
    int Numero,
    string Origen,
    string Hash,
    DateTimeOffset CapturadaEn,
    string? Comentario,
    string Estado,
    bool Estable,
    string? UsuarioSolicitante);

public sealed record VersionConfiguracionDetalle(
    long Id,
    long DispositivoId,
    string Dispositivo,
    long CapturaId,
    int Numero,
    string Origen,
    string Hash,
    DateTimeOffset CapturadaEn,
    string? Comentario,
    string Estado,
    bool Estable,
    string? UsuarioSolicitante,
    string Contenido);

public sealed record CargaArchivoResultado(
    CapturaResumen Captura,
    VersionConfiguracionResumen Version);

public sealed record CapturaRemotaResultado(
    CapturaResumen Captura,
    VersionConfiguracionResumen Version);

public sealed record ArchivoConfiguracionProcesado(
    string NombreArchivo,
    string Contenido,
    string Hash,
    string Comentario);

public sealed record ContenidoConfiguracionProcesado(
    string Contenido,
    string Hash);
